src/certgen_certificate_generator_server.h

```diff
@@ -22,8 +22,42 @@
 #ifndef CERTGEN_CERTIFICATE_GENERATOR_SERVER_H_INCLUDED
 #define CERTGEN_CERTIFICATE_GENERATOR_SERVER_H_INCLUDED
 
+#include <functional>
+#include <fty_common_sync_server.h>
+
+#define CONFIG_PATH "/etc/fty/fty-certificate-generator/"
+
 namespace certgen
 {
+    using Command   = std::string;
+
+    using FctCommandHandler = std::function<std::string (const std::vector<std::string> &)>;
+
+    class CertificateGeneratorServer final : public fty::SyncServer
+    {
+        public:
+            explicit CertificateGeneratorServer();
+            fty::Payload handleRequest(const fty::Sender & sender, const fty::Payload & payload) override;
+
+        private: // methods
+            // List of supported commands with a reference to the handler for this command.
+            std::map<Command, FctCommandHandler> m_supportedCommands;
+
+            // Handlers for all supported commands
+            std::string handleGenerateSelfsignedCertificate(const fty::Payload & params);
+            std::string handleGenerateCSR(const fty::Payload & params);
+            std::string handleImportCertificate(const fty::Payload & params);
+
+            // static helpers
+            std::list<std::string> getSystemIPs (int family);
```

Comment:
static helpers could be define as static and only in the cpp file

Reply:
Addressed.